### PR TITLE
Delete trip

### PR DESCRIPTION
**Pull request by ChrisHutchinson1982**
> save button now only appears when user is logged in
> null values are now shown as N/A in trip table
> user can now delete a trip and trips page refreshes

### CODE REVIEW

File: frontend/src/components/trip/trip.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      method: "DELETE",
+      headers: {
+        Authorization: `Bearer ${token}`,
+        trip_id: trip._id,
+      },
+    });
+
+    await response.json();
+
+    if (response.status !== 200) {
```
Comment (ThomasSel) [inline frontend/src/components/trip/trip.js:53]: Shouldn't you check the response status before calling `response.json()`?

Comment (ThomasSel) [reply frontend/src/components/trip/trip.js:53]: In fact you don't use `response.json()` at all so you probably don't need it

File: frontend/src/components/trip/trip.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                </tr>
+              </thead>
+              <tbody>
+                {unpackEmissions(trip).map((emissions) => (
+                  <tr>
+                    <td>{emissions.type}</td>
+                    <td>
+                      {emissions.plane
+                        ? `${emissions.plane.toFixed(1)}`
+                        : "N/A"}
```
Comment (ThomasSel) [inline frontend/src/components/trip/trip.js]: Could put this in a loop for clarity

**Review by ThomasSel — APPROVED** [review]
Good, especially the refactoring of the trip into a separate component!

**Comment by ChrisHutchinson1982** [comment]
Refactored to include feedback comments

**Comment by ChrisHutchinson1982** [comment]
Closed in error, merging request now.

File: frontend/src/components/trip/trip.js (hunk 2 above)
Comment (ChrisHutchinson1982) [reply frontend/src/components/trip/trip.js]: This has been updated

File: frontend/src/components/trip/trip.js (hunk 1 above)
Comment (ChrisHutchinson1982) [reply frontend/src/components/trip/trip.js:53]: This has been removed

**Pull request merged**